solutions/task_7.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python3
+"""
+TASK: Implement at least following classes to execute unittests without errors:
+
+'MixInMeta':
+    Metaclass which can be used to get ability to mix in classes.
+
+'MixInBase':
+    Class which can be inherited to get ability to mix in classes. (To compare with 'MixInMeta'.)
+
+'DocStyleMeta':
+    Metaclass which can be used to check doc-strings of methods in a class.
+
+'CacheMeta':
+    Metaclass which can be used to cache all methods in a class.
+"""
+import re
+import time
+import unittest
+import types
+from functools import wraps
+
+
+class MixInBase(object):
+
+    @classmethod
+    def mix_in(cls, new_base):
+        cls.__bases__ += (new_base,)
+
+
+class MixInMeta(type):
+
+    def mix_in(cls, new_base):
+        cls.__bases__ += (new_base,)
+
+
+class DocStyleMeta(type):
+
+    def __new__(mcs, name, bases, dct):
+        for key, value in dct.items():
+            if isinstance(value, types.FunctionType) and re.search('__', key) is None:
+                if value.__doc__ is None:
+                    raise DocStyleError("'{}' has no documentation string.".format(key))
+                if re.search('  ', value.__doc__) is not None:
+                    raise DocStyleError("'{}' has more than one space delimiter between words in documentation "
+                                        "string.".format(key))
+                if re.search('[A-Z]', str(key)) is not None:
+                    raise DocStyleError("'{}' has no snake-case style.".format(key))
+        return super().__new__(mcs, name, bases, dct)
```

Comment:
I suppose to do it explicitly or keep in mind purpose of ```super``` arguments.
```super(DocStyleMeta, mcs).__new__(mcs, name, bases, dct)```
https://docs.quantifiedcode.com/python-anti-patterns/correctness/missing_argument_to_super.html
https://docs.python.org/3.1/library/functions.html#super
P.S. Python2 will fail with this call.